### PR TITLE
properly handle read timeouts with use_poll=True

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/pexpect/pexpect/issues/491

### CODE REVIEW

**Comment by ryanpetrello** [comment]
cc @cooperlees @shoyer @takluyver

File: pexpect/utils.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def poll_ignore_interrupts(fds, timeout=None):
 
     poller = select.poll()
     for fd in fds:
-        poller.register(fd)
+        poller.register(fd, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLERR)
```
Comment (ryanpetrello) [inline pexpect/utils.py]: The prior version of this was returning immediately due to listening for `select.POLLOUT` for writes.

File: tests/test_misc.py [hunk 2]
```diff
@@ -87,6 +87,12 @@ def test_read_poll(self):
         remaining = child.read().replace(_CAT_EOF, b'')
         self.assertEqual(remaining, b'abc\r\n')
 
+    def test_read_timeout(self):
```
Comment (cooperlees) [inline tests/test_misc.py]: NIT: test_read_poll_timeout

File: pexpect/utils.py (hunk 1 above)
Comment (cooperlees) [reply pexpect/utils.py]: Great catch. Makes sense to me.

**Comment by takluyver** [comment]
Yup, makes sense to me. You're right that POLLOUT events shouldn't be important to us.

**Comment by takluyver** [comment]
Yup, give or take merging a couple of other PRs that are going on at the moment.

@cooperlees feel free to merge this one whenever you think it's had long enough for people to review it. :-)

**Comment by cooperlees** [comment]
I would, but I do not have commit/merge access!

**Comment by takluyver** [comment]
Oh, right, sorry, I forget who's a maintainer...

**Comment by takluyver** [comment]
#490 is the other one that I'll try to merge before doing a release.

**Comment by takluyver** [comment]
If anyone wants to help towards the release, a PR to start updating the release notes in the docs would be welcome. :-)

**Comment by ryanpetrello** [comment]
@takluyver https://github.com/pexpect/pexpect/pull/494

**Pull request merged**